library/asn1parse.c

```diff
@@ -8,7 +8,7 @@
 #include "common.h"
 
 #if defined(MBEDTLS_ASN1_PARSE_C) || defined(MBEDTLS_X509_CREATE_C) || \
-    defined(MBEDTLS_PSA_UTIL_HAVE_ECDSA)
+    defined(PSA_WANT_ALG_ECDSA) || defined(PSA_WANT_ALG_DETERMINISTIC_ECDSA)
```

Comment:
Needing the presence of ECDSA, but not caring which variant, is a common concern. Rather than repeat this pattern everywhere, we should introduce a macro that says that some ECDSA is available. `PSA_HAVE_ALG_SOME_ECDSA`?

Reply:
What do you think where should it be defined?

Reply:
I guess `include/psa/crypto_adjust_config_*`, since we use `adjust_config` both to define public symbols based on other public symbols, and to define internal symbols based on public symbols. None of the existing ones really fit, so it would be a new one. `crypto_adjust_config_derived.h`?

Reply:
That same file should have the definition of `PSA_WANT_ALG_SOME_PAKE`. It was added directly to `config_psa.h`, but we split that file because it was a mess, we shouldn't have added another individual symbol to it.